builder/build/module.rs

```diff
@@ -0,0 +1,8 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+
+pub trait Module {
```

Comment:
I think it would be useful to have some documentation on this trait, and on how to add a new crate to the build system. Also would it make sense to define a `new` method so that modules have the same interface, or would that be too rigid ?

Reply:
Agree on the documentation. Could you elaborate a little bit about the reasoning on adding a new method?

Reply:
This would make `main.rs` less module specific since adding a module would always be the same 
```
builder.add_module(Box::new(MyModule::new(...));
```
But that would make it more rigid having to pass useless params for some Module and making it more complicated to add new params. So probably not worth it.